libraries/AP_Arming/AP_Arming.cpp

```diff
@@ -168,10 +168,27 @@ bool AP_Arming::barometer_checks(bool report)
 {
     if ((checks_to_perform & ARMING_CHECK_ALL) ||
         (checks_to_perform & ARMING_CHECK_BARO)) {
-        if (!AP::baro().all_healthy()) {
+        const AP_Baro *baro = AP_Baro::get_singleton();
+        if (!baro->all_healthy()) {
             check_failed(ARMING_CHECK_BARO, report, "Barometer not healthy");
             return false;
         }
+        if (checks_to_perform & ARMING_CHECK_GPS) {
+            const AP_GPS *gps = AP_GPS::get_singleton();
+            if (gps->status() >= AP_GPS::GPS_OK_FIX_3D) {
+                if (gps->get_vdop() < 170 && gps->num_sats() > 8) {
```

Comment:
this is duplicating existing checks that the EKF does for the GPS

Reply:
So we can just wait for INS checks to be passed. Good idea.